### PR TITLE
Update pagination guidance to clarify the difference between server-side and client-side paging

### PR DESCRIPTION
**Pull request by corranrogue9**
I noticed that we have been considering the use of `@nextLink` to be synonymous with server-driven paging. There are some workloads that provide a `@nextLink` using the `$skip` and `$top` query parameters. These workloads are "really" using client-driven paging since they make no guarantee that elements aren't missing or duplicated. I added a section to call out that services must use *some* form of pagination, and that they can use either server-driven, client-driven, or both. I also note that server-driven paging leverages the `$skiptoken` query paramter.

### CODE REVIEW

**Comment by OlgaPodo** [comment]
I concur with these changes as an interim clarification step even though we will need to merge REST guidelines to Graph guidelines later this year. @dkershaw10 could you please review as well?

**Comment by OlgaPodo** [comment]
@corranrogue9 Could you please add an additional guidance to cover  https://github.com/microsoft/api-guidelines/issues/467 as it directly relates to paging?

**Comment by OlgaPodo** [comment]
as per discussion decided to resolve issue #467 as a separate work.

**Comment by corranrogue9** [comment]
@dkershaw10  or @OlgaPodo do you have additional feedback on this change?

**Review by dkershaw10 — APPROVED** [review]

File: Guidelines.md [hunk 1]
```diff
@@ -976,7 +978,45 @@ Content-Type: application/json
 }
 ```
 
-#### 9.8.2. Client-driven paging
+The `@nextLink` MAY be populated using either server-driven paging or client-driven paging (`@nextLink`s generated using client-driven paging should not include the `$top` query parameter).
+
+#### 9.8.2. Server-driven paging
+
+The server MAY provide server-driven paging by populating the continuation token with a `$skiptoken` query parameter.
+The `$skiptoken` value is opague for clients and its structure should not be assumed.
```
Comment (PetRich-MSFT) [inline Guidelines.md]:
```suggestion
The `$skiptoken` value is opaque for clients and its structure should not be assumed.
```

File: Guidelines.md [hunk 2]
```diff
@@ -976,7 +978,45 @@ Content-Type: application/json
 }
 ```
 
-#### 9.8.2. Client-driven paging
+The `@nextLink` MAY be populated using either server-driven paging or client-driven paging (`@nextLink`s generated using client-driven paging should not include the `$top` query parameter).
+
+#### 9.8.2. Server-driven paging
+
+The server MAY provide server-driven paging by populating the continuation token with a `$skiptoken` query parameter.
```
Comment (PetRich-MSFT) [inline Guidelines.md:985]:
```suggestion
The server MAY provide server-driven paging by populating the continuation token with a URL containing a `$skiptoken` query parameter.
```

**Review by PetRich-MSFT — APPROVED** [review]
Minor typo to be fixed

**Pull request merged**